File: work-schedule-service/src/main/java/com/averude/uksatse/scheduler/microservice/workschedule/model/CommonDataDTO.java
package com.averude.uksatse.scheduler.microservice.workschedule.model;

import com.averude.uksatse.scheduler.core.model.entity.DayType;
import com.averude.uksatse.scheduler.core.model.entity.Employee;
import com.averude.uksatse.scheduler.core.model.entity.Position;
import com.averude.uksatse.scheduler.core.model.entity.structure.Shift;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

import java.util.List;

@Getter
@RequiredArgsConstructor
public class CommonDataDTO {
    private final List<DayType> dayTypes;
    private final List<Position> positions;
    private final List<Shift> shifts;
    private final List<Employee> employees;
}

File: frontend-server/src/app/model/dto/united/common-data-dto.ts
import { DayType } from "../../day-type";
import { Shift } from "../../shift";
import { Employee } from "../../employee";
import { Position } from "../../position";

export class CommonDataDTO {
  dayTypes:  DayType[];
  positions: Position[];
  shifts:    Shift[];
  employees: Employee[];
}
